pandas/tests/generic/test_frame.py

```diff
@@ -157,27 +157,25 @@ def test_validate_bool_args(self, value):
 
         msg = 'For argument "inplace" expected type bool, received type'
         with pytest.raises(ValueError, match=msg):
-            super(DataFrame, df).rename_axis(
-                mapper={"a": "x", "b": "y"}, axis=1, inplace=value
-            )
+            super().rename_axis(mapper={"a": "x", "b": "y"}, axis=1, inplace=value)
```

Comment:
This doesn't work, I think it should be enough to just make it
```
df.copy().rename_axis(mapper={"a": "x", "b": "y"}, axis=1, inplace=value)
```
(and likewise for the others below)

I presume the idea of doing `super(DataFrame, df)` was to not modify `df` itself. It was added in https://github.com/pandas-dev/pandas/pull/14318 - cc @rforgione in case you happen to remember

Reply:
> This doesn't work, I think it should be enough to just make it
> 
> ```
> df.copy().rename_axis(mapper={"a": "x", "b": "y"}, axis=1, inplace=value)
> ```
> 
> (and likewise for the others below)
> 
> I presume the idea of doing `super(DataFrame, df)` was to not modify `df` itself. It was added in #14318 - cc @rforgione in case you happen to remember

I have fixed this now. @MarcoGorelli please review.

Reply:
yikes, this is from quite a long time ago (6 years). I vaguely remember working on this -- we were trying to fix the case where various truth-y args were interpreted as `True` to the `inplace` arg for some operations, resulting in some folks passing `0` or `"False"` and accidentally setting the arg value to `True`. However, I would have to dig into the details a bit to better understand what was going on here. I'm guessing your instincts are correct here that we were not intending to modify the df itself. But to be honest, I haven't had time to dig into the details of this specific issue. Just didn't want to leave it unresponded-to!

Reply:
hey, no worries, thanks for your response!